postprocessing/pp-opus.c

```diff
@@ -128,176 +129,29 @@ int janus_pp_opus_process(FILE *file, janus_pp_frame_packet *list, int *working)
 			last_seq = tmp->seq;
 			steps++;
 		}
-		ogg_packet *op = op_from_pkt((const unsigned char *)buffer, bytes);
-		pos = (tmp->ts - list->ts) / 48 / 20 + 1;
 		JANUS_LOG(LOG_VERB, "pos: %06"SCNu64", writing %d bytes out of %d (seq=%"SCNu16", step=%"SCNu16", ts=%"SCNu64", time=%"SCNu64"s)\n",
 			pos, bytes, tmp->len, tmp->seq, diff, tmp->ts, (tmp->ts-list->ts)/48000);
-		op->granulepos = 960*(pos); /* FIXME: get this from the toc byte */
-		ogg_stream_packetin(stream, op);
-		g_free(op);
-		ogg_write();
-		ogg_flush();
+		av_packet_unref(pkt);
+		pkt->stream_index = 0;
+		pkt->data = buffer;
+                pkt->size = bytes;
+		pkt->pts = pkt->dts = av_rescale_q(tmp->ts - list->ts, timebase, fctx->streams[0]->time_base);
```

Comment:
Is this maybe why you were getting issues with the silence packets being inserted manually as we did when using libogg directly? Does the ogg format target in ffmpeg automatically insert silence audio packets looking at what the pts is, or is still increased monotonically even in case of gaps?

Reply:
The sample I used to test has the audio lasting much longer than the video, I thought the problem was in the audio, but probably it is the video having problems.

I still need to get a local setup to get better samples.

Inserting the silence seems fine with both ogg and mkv output.

Reply:
> Inserting the silence seems fine with both ogg and mkv output.

Ack, thanks!

> I still need to get a local setup to get better samples.

There are a couple of old .mjr files you can play with in [this folder of the repo](https://github.com/meetecho/janus-gateway/tree/master/plugins/recordings): they use the old format, but should be ok if you want to test audio and video streams that have the same length.